src/rtctools_heat_network/heat_mixin.py

```diff
@@ -3736,8 +4048,8 @@ def __pipe_topology_constraints(self, ensemble_member):
                 p if p in self.hot_pipes else self.cold_to_hot_pipe(p)
             ]
             v = []
-            for pc_var_name in pipe_classes.values():
-                v.append(self.extra_variable(pc_var_name, ensemble_member))
+            for pc in pipe_classes.values():
```

Comment:
In this part of the code we set the heat_loss variable, I think for clarity it is better if we put these in a separate function, as they need to be set independently of topology optimization.

Reply:
Yes, good suggestion :)